src/charm.py

```diff
@@ -982,6 +1003,44 @@ def _generate_bessd_startup_command(self) -> str:
             hugepages_cmd = "-m 0"  # "-m 0" means that we are not using hugepages
         return f"/bin/bessd -f -grpc-url=0.0.0.0:{BESSD_PORT} {hugepages_cmd}"
 
+    def push_file(
+        self,
+        container: Container,
+        path: Union[str, PurePath],
+        source: str,
+    ) -> None:
+        """Pushes source content to path in container.
+
+        Args:
+            container: Container object
+            path: Path in which content is pushed
+            source: Content to be pushed to container
+        """
+        try:
+            container.push(path=path, source=source)
+        except ConnectionError:
+            self.unit.status = WaitingStatus(f"Waiting for {container.name} to be ready")
+
+    def path_exists(
+        self,
+        container: Container,
+        path: Union[str, PurePath],
+    ) -> bool:
+        """Returns existence of path in container.
+
+        Args:
+            container: Container object
+            path: Path to verify the existence of
+
+        Returns:
+            bool: existence of path in container
+        """
+        try:
+            return container.exists(path=path)
+        except ConnectionError:
+            self.unit.status = WaitingStatus("Waiting for Pebble API to be ready")
```

Comment:
I don't think we should handle status in these methods, it's strange that they do this as a side effect. In the future we'll want to place all of this in a collect status handler but even for the time being I don't know that this is the right place to handle status.

Reply:
I see. At this point I can propose not to create wrappers methods, but using try-except blocks in methods performing container operations and handling the status there. It is suboptimal (as we are introducing some duplication), but we already know it's a temporary solution until we start using 'collect_status`. Wdyt?

Reply:
Could we implement the changes to use Ops.CollectStatus event first before that change ? It will help us to simplify the things.

Reply:
@gatici I was looking into that, but I don't think it changes anything. The main issue I see is that we are setting the unit status from methods which have no information on the event (hence we can't use the `add_status`). See [pgbouncer-k8s-operator](https://github.com/canonical/pgbouncer-k8s-operator/blob/af08e116c18321f3b9e1c93a63f13219f8c69b9b/src/charm.py#L585) which I used as guideline. Any input is really appreciated.